invenio_users_resources/resources/groups/resource.py

```diff
@@ -52,17 +52,23 @@ def search(self):
     @response_handler()
     def read(self):
         """Read a group."""
+        id_ = resource_requestctx.view_args["id"]
+        if not GroupAggregate.get_record_by_name(id_):
```

Comment:
We try to avoid to use `records` api directly in the resource layer. The best here is probably to try/except given that the `service` is [already checking that](https://github.com/inveniosoftware/invenio-users-resources/blob/master/invenio_users_resources/services/users/service.py#L77).
```
try:
    item = self.service.read(...
except LookupError:
    abort(404)
```